hmcts/pre-api
# S28-2256: Users have Ts and Cs accepted at timestamp

### Change description

- Added `terms_accepted_at` column to `portal_access` and `app_access` tables
- When a user is created this will be null
- 2 new `POST` endpoints to accept the ts and cs for the authenticated user in the `X-User-Id` header
    - `/users/acceptPortalTerms`
    - `/users/acceptAppTerms`
- The access DTOs now have `terms_accepted_at` and `is_terms_accepted`
- `is_terms_accepted` is a boolean and withh be false if `terms_accepted_at` is null or if the timestamp is more that 1 year ago



> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

- Create new portal user
- Send `GET` request to `/users/by-email/{email}`
- Verify `is_terms_accepted` is false
- Send `POST` request to `/users/acceptPortalTerms` with the new user as the `X-User-Id` header
- Send `GET` request to `/users/by-email/{email}`
- Verify `is_terms_accepted` is true


- Create new app user
- Send `GET` request to `/users/{userId}`
- Verify `is_terms_accepted` is false
- Send `POST` request to `/users/acceptAppTerms` with the new user as the `X-User-Id` header
- Send `GET` request to `/users/{userId}`
- Verify `is_terms_accepted` is true
